src/deferred.js

```diff
@@ -13,6 +13,23 @@ function Thrower( ex ) {
 	throw ex;
 }
 
+function adoptValue( value, resolve, reject ) {
+	var method;
+
+	// Check for promise aspect first to privilege synchronous behavior
+	if ( value && jQuery.isFunction( ( method = value.promise ) ) ) {
```

Comment:
Promises/A+ requires accessing `.then` only once; `if ( jQuery.isFunction( value.then ) ) { value.then.call(…); }` is prohibited (cf. [Notes 3.5](https://promisesaplus.com/#point-75)).